### PR TITLE
Support collections editing and new DS type

### PR DESCRIPTION
**Pull request by blokhin**
@knopki `client.v0.collections.create()` also supports editing the existing collections (the `id` has to be supplied), see https://github.com/basf/metis-bff/blob/f6f73257d6f2603496118676969663e4d6e4deaa/routes/%5Bversion%5D/collections/_helpers.js#L65

However here is a caveat: 3 kwargs `typeId`,  `dataSources`, and `users` are mapped onto `type_id`, `data_source_ids`, and `user_ids` (note camel case vs. snake case) which prevents me from doing `client.v0.collections.create(**dict(metis_collection_changed))`. I assume, the `MetisCollectionCreateDTO` should be upgraded for that, right?

### CODE REVIEW

**Comment by blokhin** [comment]
TODO setting `id=None` for creating a new collection

File: metis_client/dtos/collection.py [hunk 1]
```diff
@@ -30,8 +33,9 @@ class MetisCollectionTypeDTO(MetisTimestampsDTO):
 
 
 class MetisCollectionCommonDTO(TypedDict):
-    "Common fields of collection's DTOs"
+    "Common fields of collection DTOs"
 
+    id: int
```
Comment (knopki) [inline metis_client/dtos/collection.py:38]: Remove or `id: NotRequired[int]` here, not at `MetisCollectionCreateDTO`.

File: metis_client/dtos/datasource.py [hunk 2]
```diff
@@ -18,16 +18,16 @@
 
 
 class DataSourceType(int, Enum):
-    "Data source types"
+
```
Comment (knopki) [inline metis_client/dtos/datasource.py:21]: Why docstring remove?

File: metis_client/namespaces/v0_collections.py [hunk 3]
```diff
@@ -48,6 +49,9 @@ async def create_event(
             users=opts.get("user_ids", []),
             visibility=opts.get("visibility", "private"),
         )
+
+        if opts.get("id"): payload["id"] = opts["id"] # FIXME???
```
Comment (knopki) [inline metis_client/namespaces/v0_collections.py:53]: Maybe
```python
        if "id" in opts:
            payload["id"] = opts["id"]
```

**Review by knopki — CHANGES_REQUESTED** [review]
Overall, it's good PR. But fix style, please.

**Comment by knopki** [comment]
> However here is a caveat: 3 kwargs `typeId`, `dataSources`, and `users` are mapped onto `type_id`, `data_source_ids`, and `user_ids` (note camel case vs. snake case) which prevents me from doing `client.v0.collections.create(**dict(metis_collection_changed))`. I assume, the `MetisCollectionCreateDTO` should be upgraded for that, right?

Or `MetisCollectionsCreateKwargs` and `create_event(...)`? Snake case looks like a bug.

File: metis_client/dtos/datasource.py (hunk 2 above)
Comment (blokhin) [reply metis_client/dtos/datasource.py:21]: because it doesn’t bring any valuable information here

**Comment by blokhin** [comment]
The different case originates from the BFF API, which being in JavaScript contradicts Python style. How could we map fields nicely onto each other so that one doesn’t have to copy them manually? Maybe to implement constructor for the typed dict?

File: metis_client/dtos/datasource.py (hunk 2 above)
Comment (knopki) [reply metis_client/dtos/datasource.py:21]: Correct. So there are two ways to go about it: write valuable docstring or turn off the linter rule [missing-class-docstring](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/missing-class-docstring.html).

**Comment by knopki** [comment]
BFF API already mapped (manually) to Python-style DTOs in client code. Why copy manually?

> Maybe to implement constructor for the typed dict?

`TypedDict` classes can contain only type annotation. `TypedDict` is a thin type. If you need a constructor or an additional method of a class or instance, it is better to use `dataclass`.

**Comment by blokhin** [comment]
> BFF API already mapped (manually) to Python-style DTOs in client code. Why copy manually?

Nope. Otherwise as said I would be able to do `client.v0.collections.create(**dict(metis_collection_changed))` but not it’s impossible.

**Comment by blokhin** [comment]
@knopki how do I better implement `client.v0.collections.create(**dict(metis_collection_changed))` ?

**Comment by knopki** [comment]
> @knopki how do I better implement `client.v0.collections.create(**dict(metis_collection_changed))` ?

I recommend correcting the stylistic errors in this PR (`black`, `pylint`) and merge it. Then discuss the problem with dictionaries, as it can be quite large.

**Pull request merged**